sympy/simplify/cse_main.py

```diff
@@ -416,12 +416,23 @@ def cse(exprs, symbols=None, optimizations=None, postprocess=None,
     reduced_exprs : list of sympy expressions
         The reduced expressions with all of the replacements above.
     """
-    from sympy.matrices import Matrix
+    from sympy.matrices import Matrix, SparseMatrix
 
     # Handle the case if just one expression was passed.
     if isinstance(exprs, Basic):
         exprs = [exprs]
 
+    copy = exprs
+    temp = []
+    for e in exprs:
+        if isinstance(e, Matrix) or isinstance(e, SparseMatrix):
```

Comment:
So first I should do something like this:
```
temp.append(e.__class__(x.rows, x.cols, {}))
```
followed by filling the required spaces through the for loop you wrote
```
for k,v in e[2]:
...  c[-1][k] = v
```
right ?